draft-ietf-quic-http.md

```diff
@@ -297,7 +297,7 @@ of informational (1xx) HTTP responses (see {{!RFC7230}}, Section 3.2 and
 PUSH_PROMISE frames (see {{frame-push-promise}}) MAY be interleaved with the
 frames of a response message indicating a pushed resource related to the
 response. These PUSH_PROMISE frames are not part of the response, but carry the
-headers of a separate HTTP request message.  See {{server-push}} for more
+headers of separate HTTP request messages.  See {{server-push}} for more
```

Comment:
I think this falls into the trap that HTTP/QUIC PUSH_PROMISE frames can be sent multiple times. Was the original text supposed to capture that design feature? The new text seems to lose that description (PUSH_PROMISE for different messages) but accommodating both might get wordy.

Reply:
In context, I don't think it says anything about PUSH_PROMISE frames that are sent multiple times on different streams.  It's talking about a single response, and says that the PUSH_PROMISE frames carry headers of separate requests being pushed.  If the same PUSH_PROMISE shows up during another response, that's mostly independent.

Can you suggest wording you think would be clearer?

Reply:
My original point is that you could send several PUSH_PROMISE frames, for the same promised resource with the same PUSH_ID, on a single request stream.

Something simpler could be to take away all mention of PUSH_PROMISE headers in th:

A server MAY chose to interleave one or more PUSH_PROMISE frames (see {{frame-push-promise}}) with the response message. However, these are not part of the response.  Server push is described further in {{server-push}}.

Reply:
Works for me.  (But why *would* you send several PUSH_PROMISEs for the same resource on the same request stream?)

Reply:
I couldn't slip that passed you, could I? :)

I wouldn't expect conventional HTTP/QUIC servers to do this. However, the existing specification allows for this and I would hope that implementations can implement the required mechanics.

As for a reason, in the absence of a true QUIC transport-level FEC, we could imagine an application-level FEC that repeats important (and relatively cheap) metadata. Its true that today streams are reliable but some form of unreliable streams may benefit from a smarter repetition strategy. 

Independent PUSH_PROMISE frames may duplicate the same data. Request and response header messages could be FEC'd by purposefully sending duplicate STREAM frames (same offset and content) containing such metadata. 

Both forms of repetition could work together.